src/perform_step/extrapolation_perform_step.jl

```diff
@@ -15,30 +15,48 @@ end
 
 function perform_step!(integrator,cache::AitkenNevilleCache,repeat_step=false)
   @unpack t,dt,uprev,u,f,p = integrator
-  @unpack k,fsalfirst,T,utilde,atmp,dtpropose,cur_order,A = cache
+  @unpack k,fsalfirst,T,utilde,atmp,dtpropose,cur_order,A,tmps = cache
 
   @muladd @.. u = uprev + dt*fsalfirst
 
   T[1,1] = copy(u)
 
-  halfdt = dt/2
-  for i in 2:size(T)[1]
-    # Solve using Euler method
-    @muladd @.. u = uprev + halfdt*fsalfirst
-    f(k, u, p, t+halfdt)
-    integrator.destats.nf += 1
-    for j in 2:2^(i-1)
-      @muladd @.. u = u + halfdt*k
-      f(k, u, p, t+j*halfdt)
+  if integrator.alg.threading == false
+    for i in 2:size(T)[1]
+      dt_temp = dt/(2^(i-1))
+      # Solve using Euler method
+      @muladd @.. u = uprev + dt_temp*fsalfirst
+      f(k, u, p, t+dt_temp)
       integrator.destats.nf += 1
+      for j in 2:2^(i-1)
+        @muladd @.. u = u + dt_temp*k
+        f(k, u, p, t+j*dt_temp)
+        integrator.destats.nf += 1
+      end
+      T[i,1] = copy(u)
+      # Richardson Extrapolation
+      for j in 2:i
+        T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
+      end
     end
-    T[i,1] = copy(u)
-    # Richardson Extrapolation
-    for j in 2:i
-      T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
+  else
+    Threads.@threads for i in 2:size(T)[1]
+      dt_temp = dt/(2^(i-1))
+      # Solve using Euler method
+      @muladd @.. tmps[i] = uprev + dt_temp*fsalfirst
```

Comment:
Instead of making this per `i`, should be per thread. You can use `threadid` to know which thread it is, and `nthreads` to know how many threads you have available for when you preallocate `tmps`.